falcon/app.py

```diff
@@ -461,10 +462,28 @@ def add_middleware(self, middleware: object) -> None:
         #   the chance that middleware may be None.
         if middleware:
             try:
-                self._unprepared_middleware += middleware
+                middleware = list(middleware)  # type: ignore
```

Comment:
Bonus points if anyone knows how type this correctly. Since we optionally allow passing in any `object`, `mypy` complains:
```
No overload variant of "list" matches argument type "object"  [call-overload]
```

Reply:
I don't think you can. Maybe you can use
```py
try:
  if TYPE_CHECKING:
    assert isinstance(middleware, Iterable)
  ...
```

But I'm not sure it's worth it